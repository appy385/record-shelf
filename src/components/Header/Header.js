import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';

const Header = () => (
  <div className="header">
    <Link to="/">
      <span className="header-shelf">My</span>
      <span className="header-shelf"><strong>Record</strong></span>
      <span className="header-shelf">Shelf</span>
    </Link>
  </div>
);

export default Header;
