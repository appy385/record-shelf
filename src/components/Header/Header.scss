.header{
    display: flex;
    background-color: #38497E;
    padding: 1rem;
}
.header>a {
    text-decoration: none;
}
.header-shelf{
    padding-right: 0.3rem;
    font-size: 1.2rem;
    color: white;
}