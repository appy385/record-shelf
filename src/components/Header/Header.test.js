import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

describe(Header.name, () => {
  test('should match snapshot for theme light', () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
      ,
    );
    expect(container).toMatchSnapshot();
  });
  test('should go to / route when My/record/shelf is clicked', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
      ,
    );
    const syncLink = screen.getByText('My');
    fireEvent.click(syncLink);
    expect(document.location.href).toBe('http://localhost/');
  });
});
