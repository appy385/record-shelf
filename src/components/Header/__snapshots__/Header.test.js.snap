// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot for theme light 1`] = `
<div>
  <div
    class="header"
  >
    <a
      href="/"
    >
      <span
        class="header-shelf"
      >
        My
      </span>
      <span
        class="header-shelf"
      >
        <strong>
          Record
        </strong>
      </span>
      <span
        class="header-shelf"
      >
        Shelf
      </span>
    </a>
  </div>
</div>
`;
